web/hooks/use-user.ts

```diff
@@ -145,6 +140,28 @@ export const useUserRecommendedMarkets = (userId = '_', count = 500) => {
     ['user-viewed-markets', userId, count],
     viewedMarketsQuery
   )
+
+  // Filter repeatedly seen market cards within the last week
+  const viewedMarketCardsQuery = query(
+    collection(db, `users/${userId}/events`),
+    where('name', '==', 'view market card'),
+    where('timestamp', '>', Date.now() - 7 * DAY_MS),
+    orderBy('timestamp', 'desc'),
+    limit(count)
+  ) as Query<ContractCardView>
+  const viewedMarketCardEvents = useFirestoreQueryData<ContractCardView>(
+    ['user-recently-viewed-market-cards', userId, count],
+    viewedMarketCardsQuery
+  )
+  const viewedMarketCardIds =
+    viewedMarketCardEvents.data?.map((e) => e.contractId) ?? []
+  // get the count the number of times each market card was viewed
+  const marketCardViewCounts = countBy(viewedMarketCardIds)
+  // filter out markets that were viewed 3 times
+  const viewedMultipleTimesMarketIds = uniq(viewedMarketCardIds).filter(
```

Comment:
maybe change to 4 or 5 times? I kind of like seeing cards a few times before I'm bored of them.

Reply:
Not sure. But I bet there's some better way eventually than just a hard cutoff after a certain number of views.